runtime/common/src/crowdloan.rs

```diff
@@ -601,6 +594,27 @@ impl<T: Config> Module<T> {
 	pub fn crowdloan_kill(index: TrieIndex) -> child::KillChildStorageResult {
 		child::kill_storage(&Self::id_from_index(index), Some(T::RemoveKeysLimit::get()))
 	}
+
+	/// This function checks all conditions which would qualify a crowdloan has ended.
+	/// * If we have reached the `fund.end` block OR the first lease period the fund is
+	///   trying to bid for has started already.
+	/// * And, if the fund has enough free funds to refund full raised amount.
+	fn ensure_crowdloan_ended(
```

Comment:
This refactoring is in anticipation of me writing another function using this same logic, and not wanting to repeat it twice.

Reply:
The function would allow for automatic refunds, so that users do not need to make lots of separate calls.